src/interface/swarm_comms.cpp

```diff
@@ -236,42 +240,102 @@ void Swarm::LoadBuffers_() {
   auto &y = Get<Real>(swarm_position::y::name()).Get();
   auto &z = Get<Real>(swarm_position::z::name()).Get();
 
-  // Zero buffer index counters
-  auto &buffer_counters = buffer_counters_;
-  pmb->par_for(
-      PARTHENON_AUTO_LABEL, 0, NMAX_NEIGHBORS - 1,
-      KOKKOS_LAMBDA(const int n) { buffer_counters[n] = 0; });
+  if(max_active_index_ >= 0) {
+    // Make an n particle sized array of index, buffer pairs (with SwarmKey struct)
+    ParArray1D<SwarmKey> buffer_sorted("buffer_sorted", max_active_index_+1);
```

Comment:
This PR reuses the `SwarmKey` struct that used to sort and count particles in a cell. I can make a new version of that struct with more appropriate member names if you'd like.

Reply:
Ditto for the `buffer_start` just below this comment (although it won't need to be resized as the swarm pool increases)

Reply:
I've added these to the class data and changed the `cell_idx_1d_` name to `sort_idx_`. Thanks!